st2common/st2common/models/system/paramiko_command_action.py

```diff
@@ -47,7 +47,8 @@ def get_full_command_string(self):
             command = 'sudo %s -- bash -c %s' % (sudo_arguments, command)
 
             if self.sudo_password:
-                command = 'echo -e %s | %s' % (quote_unix('%s\n' % (self.sudo_password)), command)
+                command = ('set +o history ; echo -e %s | %s' %
```

Comment:
Now my understanding of the use of `+` here means you're enabling history, but since we're exporting `HISTFILE=/dev/null` it results in nothing being logged. Sort of a way of forcing the logging to happen on our terms (into the void). I have that right?

Reply:
Afaik, it's turned on with `+o`, but I also additionally set HISTFILE and HISTSIZE just in case (you can also test `set +o history` locally, you will see this will be the last command saved in history until turning it back on with `-o`).

And I do agree, I would also expect it to work in the opposite manner, so the "API" is kinda confusing.